xds/src/main/java/com/linecorp/centraldogma/xds/k8s/v1/XdsKubernetesEndpointFetchingService.java

```diff
@@ -134,15 +136,24 @@ protected void handleXdsResource(String path, String contentAsText, String group
             oldUpdater.close();
         }
         final KubernetesEndpointsUpdater updater =
-                new KubernetesEndpointsUpdater(commandExecutor, kubernetesEndpointGroup, executorService,
+                new KubernetesEndpointsUpdater(commandExecutor, future, executorService,
                                                groupName, watcherName, endpointWatcher.getClusterName());
         updaters.put(watcherName, updater);
-        kubernetesEndpointGroup.addListener(endpoints -> {
-            if (endpoints.isEmpty()) {
-                return;
+        future.handle((kubernetesEndpointGroup, cause) -> {
+            if (cause != null) {
```

Comment:
Would you leave a warning log for `cause`? I guess https://github.com/line/centraldogma/pull/1038/files#diff-131baeee793bf759dddfe5493e4e204e04f40869576f3d654124581c973114beR202-R205 exception is neither propagated nor logged.

Reply:
That's a good suggestion. 👍 I'll add a log for that.
Initially, I didn’t take any action because the KubernetesEndpointGroup is already created in XdsClusterService without any issues, so I assumed the exception was a rare occurrence. However, it's still important to log it. 😉